presto-main/src/main/java/com/facebook/presto/execution/TaskManagerConfig.java

```diff
@@ -44,10 +44,12 @@
     private DataSize maxIndexMemoryUsage = new DataSize(64, Unit.MEGABYTE);
     private boolean shareIndexLoading;
     private int maxWorkerThreads = Runtime.getRuntime().availableProcessors() * 2;
+    private Integer minWorkerThreads;
```

Comment:
Message:

TaskExecutorController now automatically adjusts the number of runner threads within a specified range to meet a target CPU utilization. Proportional–integral–derivative (PID) controller with conservative parameters is used to control the adjustment process.